libpius/mailer.py

```diff
@@ -188,13 +200,14 @@ def verify_pass(self):
         # So, in order be secure on older pythons we ehlo() and then check the
         # response before attempting startls.
         try:
-            smtp.ehlo()
-            if not smtp.has_extn("STARTTLS"):
-                # Emulate 2.6 behavior
-                raise smtplib.SMTPException("Server does not support STARTTLS")
-            smtp.starttls()
-            # must ehlo after startls
-            smtp.ehlo()
+            if not self.ssl:
+                smtp.ehlo()
+                if not smtp.has_extn("STARTTLS"):
+                    # Emulate 2.6 behavior
+                    raise smtplib.SMTPException("Server does not support STARTTLS")
+                smtp.starttls()
+                # must ehlo after startls
+                smtp.ehlo()
```

Comment:
Python [smtplib's doc](https://docs.python.org/3/library/smtplib.html#smtplib.SMTP.ehlo) says: "Unless you wish to use [has_extn()](https://docs.python.org/3/library/smtplib.html#smtplib.SMTP.has_extn) before sending mail, it should not be necessary to call this method explicitly."

Since we are only using `has_extn()` to check for STARTTLS support, it should not be needed on SSL.

Reply:
copy that.